src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,16 +193,21 @@ export default class DefaultMessagingSession implements MessagingSession {
     }
   }
 
+  private retryConnection(): boolean {
+    return this.reconnectController.retryWithBackoff(async () => {
+      try {
+        await this.startConnecting(true);
+      } catch (e) {
+        // getMessagingSessionEndpoint can fail, so need to retry here
+        this.retryConnection();
```

Comment:
Should there be a threshold on how long one can retry connecting? I think the threshold is handled by `retryWithBackoff` already so it should execute only when the `retryConnection` returns `true` right?

I think this should be return